django-cloudlaunch/baselaunch/view_helpers.py

```diff
@@ -86,6 +87,13 @@ def get_credentials_from_request(cloud, request):
                     }
         else:
             return {}
+    elif isinstance(cloud, models.GCE):
+        gce_credentials_json = request.META.get('HTTP_CL_GCE_CREDENTIALS_JSON')
```

Comment:
Ah good catch, I wasn't thinking about CR/LFs.  I think you're right and this will work as long as the JSON doesn't have CR/LFs.

The other approach would be to split out the fields into separate headers.  I considered doing this because I read somewhere that web servers tend to support headers with values only up to 8KB. But the credentials file I have is only 2,330 bytes, so it should fit comfortably.

Eventually I decided to just treat the credentials as a kind of opaque blob for the sake of simplicity. But I'd be curious to know if you have an opinion on whether I should have separate headers for each field or not.